### PR TITLE
Provide way to disable Multi Cursor (Implement #2663)

### PR DESCRIPTION
**Pull request by JayP16**
Implemented MultiCursor disable function. Now can disable the multi cursor the same way as Cursor; using "cursor.active = False" 
( #2663)

### CODE REVIEW

**Comment by tacaswell** [comment]
Could you move the `self.active` and ignore logic up to the `Widget` base-class?  I think that would be a more-general solution (and remove duplicate code).

If you are feeling ambitious, making sure that all of the sub-classes make proper use of `super` in the ignore calls is also probably a good thing to do.

**Comment by JayP16** [comment]
Done. Moved all the active logic up to the Widget class. And removed the basic ignore functions from the subclasses.

**Comment by tacaswell** [comment]
Great.

I am a bit wary of the class level attribute which it silently made an instance attribute when you assign to it, but do anything about that until someone else weighs in.

attn @efiring @mdboom @blink1073

**Comment by tacaswell** [comment]
also attn @WeatherGod

**Comment by blink1073** [comment]
IMO `MultiCursor` should subclass `AxesWidget` and leave `Widget` alone.

**Comment by JayP16** [comment]
>IMO MultiCursor should subclass AxesWidget and leave Widget alone.

That was my initial thoughts as well. But the AxesWidget takes in only single axis and all the child classes only require one so this would be the odd one out. It would also require modifications in all the subclasses of AxesWidget to check type; more ambition! Or perhaps for the single axes case make a single element tuple? But still requires modification to the subclasses.

**Comment by tacaswell** [comment]
I was in the process of typing out what @JayP16 said.

**Comment by efiring** [comment]
@tacaswell, you are referring to the _active attribute?  I don't see any problem with this.  It seems to me that this idiom makes the code more compact and makes it easier to see what the attributes are.  What is it that makes you wary of it?

**Comment by WeatherGod** [comment]
I have to agree with @tacaswell, class-level attributes turned into
instance attributes hurts my brain

**Comment by WeatherGod** [comment]
oh, wow... I never noticed that there were those other class-level attributes! That is unfortunate, and I now need an excedrin.

**Comment by tacaswell** [comment]
@efiring Yes.  It is is part as Ben says a source of headaches/confusion.  If _seems_ like a nice way to do default state, but it _might_ open up a can of worms with mutable values (which `_active` is not).  In general any prospect of unintentionally shared state makes we wary.

**Comment by efiring** [comment]
@WeatherGod Why?  I still don't understand the problem.  Does this idiom introduce some risk?  Is it any harder to understand than, say, properties?  It seems straightforward to me, and something that can enhance rather than reduce readability.  I think it is quite common.  It is similar to the situation where module-level variables turn into local variables via assignment in a function.  That's actually something I try to avoid, but in the case of classes, it seems natural to me.

**Comment by WeatherGod** [comment]
do class-level attributes get inherited upon subclassing? I thought they didn't.

**Comment by efiring** [comment]
@tacaswell,  OK, so it requires a bit of care, to avoid mutable variables.

**Comment by efiring** [comment]
I was just looking at http://ioam.github.io/param/index.html, which depends on this type of behavior, including the inheritance.  I think traitlets also use this idiom of class-level initialization.  I need to look.

**Comment by tacaswell** [comment]
Those are a bit different though as they are using meta-class magic to turn the class level attributes into instance level properties, right?

**Comment by efiring** [comment]
At least in params, the conversion into instance attributes happens only when a value is changed, so it is very much like the simple case.  I don't know how traitlets handles them.

**Comment by efiring** [comment]
It looks like HasTraits initialization causes immediate instance-level initialization of all Traits, so this is handled differently than in params.

**Comment by JayP16** [comment]
Any suggestions on what should be done? It would be problematic if it were mutable; which _active is not, like @tacaswell mentioned. Do we want a initialize method in widgets; can we see a use for it in the future?

**Comment by tacaswell** [comment]
Sorry, this PR discussion went off on a tangent (part of a rolling conversation @efiring and I are having).  Eric convinced me that it is fine as implemented.

File: lib/matplotlib/widgets.py [hunk 1]
```diff
@@ -1061,6 +1061,8 @@ def disconnect(self):
 
     def clear(self, event):
         """clear the cursor"""
+        if self.ignore(event):
```
Comment (tacaswell) [inline lib/matplotlib/widgets.py]: Clear should probably work even if the widget is not active

Comment (tacaswell) [inline lib/matplotlib/widgets.py]: If this behavior is consistent with the other widgets then disregard my last comment.

File: lib/matplotlib/widgets.py [hunk 2]
```diff
@@ -1099,7 +1103,7 @@ def _update(self):
         else:
 
             self.canvas.draw_idle()
-
+
```
Comment (tacaswell) [inline lib/matplotlib/widgets.py]: Please remove this whitespace.

**Comment by tacaswell** [comment]
Only other comment is that it might be worth starting the deprecation process on the `get/set_active` methods in favor of the just using the property, but that is probably out of scope for this PR.

**Comment by tacaswell** [comment]
Can you also add an entry into https://github.com/matplotlib/matplotlib/tree/master/doc/users/whats_new It should be documented that all `Widget` sub-classes now have these functions.

**Comment by tacaswell** [comment]
ping @JayP16 re whats_new entry.

**Comment by JayP16** [comment]
Sorry for the delay. It's been created. :)

**Comment by tacaswell** [comment]
@JayP16 Thank you!

**Pull request merged**